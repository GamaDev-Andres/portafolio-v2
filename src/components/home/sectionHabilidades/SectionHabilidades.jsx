import TitleSection from '../../TitleSection'
import Habilidad from './Habilidad'

const SectionHabilidades = () => {
  return (
    <section>
      <TitleSection textInSpan={'Mis'}>Habilidades</TitleSection>
      <div className="flex flex-col gap-2">
        <article className="flex flex-col gap-2">
          <div className="flex flex-col gap-2">
            <Habilidad porcentaje="w-[80%]">Desarrollo front-end</Habilidad>
            <Habilidad porcentaje="w-[70%]">Desarrollo full-stack</Habilidad>
            <Habilidad porcentaje="w-[70%]">Desarrollo web</Habilidad>
            <Habilidad porcentaje="w-[60%]">Desarrollo back-end</Habilidad>
          </div>
        </article>
      </div>
    </section>
  )
}

export default SectionHabilidades
