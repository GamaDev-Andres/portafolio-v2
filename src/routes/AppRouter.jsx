import { useState } from 'react';
import { BrowserRouter, Route, Routes } from 'react-router-dom';
import Home from '../components/home/Home';
import Project from '../components/project/Project';

function AppRouter() {
  return (
    <BrowserRouter>
      <Routes>
        <Route path="/" element={<Home />} />
        <Route path="/project/:id" element={<Project />} />
      </Routes>
    </BrowserRouter>
  );
}

export default AppRouter;
