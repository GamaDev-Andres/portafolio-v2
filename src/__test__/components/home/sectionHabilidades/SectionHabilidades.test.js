/**
 * @jest-environment jsdom
 */
import Enzyme from 'enzyme'
import { HelmetProvider } from 'react-helmet-async'
import { MemoryRouter } from 'react-router-dom'
import SectionHabilidades from '../../../../components/home/sectionHabilidades/SectionHabilidades'
import AppRouter from '../../../../routes/AppRouter'

describe('probando <SectionHabilidades/>', () => {
  const wrapper = Enzyme.mount(
    <SectionHabilidades />
  )
  test('debe montarse correctamente', () => {
    expect(wrapper).toMatchSnapshot()
    expect(wrapper.find('TitleSection').length).toBe(1)
    expect(wrapper.find('Habilidad').exists()).toBe(true)
  })
  test('No debe montarse en ruta "project"', () => {
    const wrapper = Enzyme.mount(
      <HelmetProvider>
        <MemoryRouter initialEntries={['/project/BARBERAPP']}>
          <AppRouter />
        </MemoryRouter>
      </HelmetProvider>
    )
    expect(wrapper.find('SectionHabilidades').exists()).toBe(false)
  })
})
