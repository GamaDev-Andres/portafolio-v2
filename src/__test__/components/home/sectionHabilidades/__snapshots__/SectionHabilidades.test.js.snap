// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`probando <SectionHabilidades/> debe montarse correctamente 1`] = `
<section
  className="sm:scroll-m-24 md:scroll-m-14"
  id="habilidades"
>
  <h2
    className="my-4 uppercase font-bold text-3xl md:text-center"
  >
    <span
      className="text-blue-400"
    >
      Mis
    </span>
     
    Habilidades
  </h2>
  <div
    className="flex flex-col gap-2"
  >
    <article
      className="flex flex-col gap-2"
    >
      <div
        className="flex flex-col gap-2"
      >
        <div>
          <span
            className="font-medium"
          >
            Desarrollo front-end
          </span>
          <div
            className="bg-fondos w-full  flex  rounded-full"
          >
            <div
              className="bg-letra  w-[80%] rounded-full p-1"
            >
              <span
                className="text-white w-full text-center block "
              >
                80
                 %
              </span>
            </div>
          </div>
        </div>
        <div>
          <span
            className="font-medium"
          >
            Desarrollo full-stack
          </span>
          <div
            className="bg-fondos w-full  flex  rounded-full"
          >
            <div
              className="bg-letra  w-[70%] rounded-full p-1"
            >
              <span
                className="text-white w-full text-center block "
              >
                70
                 %
              </span>
            </div>
          </div>
        </div>
        <div>
          <span
            className="font-medium"
          >
            Desarrollo web
          </span>
          <div
            className="bg-fondos w-full  flex  rounded-full"
          >
            <div
              className="bg-letra  w-[70%] rounded-full p-1"
            >
              <span
                className="text-white w-full text-center block "
              >
                70
                 %
              </span>
            </div>
          </div>
        </div>
        <div>
          <span
            className="font-medium"
          >
            Desarrollo back-end
          </span>
          <div
            className="bg-fondos w-full  flex  rounded-full"
          >
            <div
              className="bg-letra  w-[60%] rounded-full p-1"
            >
              <span
                className="text-white w-full text-center block "
              >
                60
                 %
              </span>
            </div>
          </div>
        </div>
      </div>
    </article>
  </div>
</section>
`;
